ops/internal/manage/depsets.go

```diff
@@ -0,0 +1,243 @@
+package manage
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/superchain-registry/ops/internal/config"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+var (
+	errInvalidActivationTime = errors.New("activation_time is before interop_time")
+	errDepsetLengths         = errors.New("inconsistent depset lengths")
+	errInconsistentDepsets   = errors.New("inconsistent depset values")
+	errMissingAddress        = errors.New("missing address")
+	errDuplicateChainIndex   = errors.New("duplicate chain index")
+)
+
+type DepsetChecker struct {
+	lgr             log.Logger
+	addrs           config.AddressesJSON
+	diskChainCfgs   map[uint64]DiskChainConfig
+	processedChains map[uint64]bool
+	chainsProcessed int
+}
+
+func NewDepsetChecker(logger log.Logger, cfgs []DiskChainConfig, addrs config.AddressesJSON) (*DepsetChecker, error) {
+	dc := &DepsetChecker{
+		lgr:             logger,
+		addrs:           addrs,
+		diskChainCfgs:   make(map[uint64]DiskChainConfig),
+		processedChains: make(map[uint64]bool),
+	}
+
+	for _, cfg := range cfgs {
+		dc.diskChainCfgs[cfg.Config.ChainID] = cfg
+	}
+
+	logger.Info("loaded depset checker instance with chain configs", "numDiskCfgs", len(cfgs))
+	return dc, nil
```

Comment:
Is there any point returning an error if it is always `nil`?

Reply:
Removed in https://github.com/ethereum-optimism/superchain-registry/pull/1013/commits/bbb7f18834dfaa2f0183640e659280aad20e2ebb